lib/matplotlib/dates.py

```diff
@@ -201,6 +220,65 @@ def _get_rc_timezone():
     MO, TU, WE, TH, FR, SA, SU)
 WEEKDAYS = (MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, SATURDAY, SUNDAY)
 
+# default epoch: passed to np.datetime64...
+_epoch = None
+_epoch_used = False
+
+
+def _reset_epoch():
+    """
+    Reset the Matplotlib date epoch to its rcParams value.
+    """
+    global _epoch
+    global _epoch_used
+
+    _epoch_used = False
+    _epoch = matplotlib.rcParams['date.epoch']
+
+
+def set_epoch(epoch):
+    """
+    Set the epoch (origin for dates) for datetime calculations.  The
+    default epoch is '1970-01-01T00:00:00'. The `.date2num` method returns
+    floating point days from the epoch, and dates that are too far
+    from the epoch will experience floating point round off errors.
+
+    This must be called before any dates are converted (i.e. right after
+    import of `.matplotlib.dates`) or a RuntimeError will be raised.
+
+    See :doc:`/gallery/ticks_and_spines/date_precision_and_epochs` for a
+    discussion.
+
+    Parameters
+    ----------
+    epoch : str
+        valid UTC date parsable by `numpy.datetime64` (do not include
+        timezone).
+
+    """
+    global _epoch
+    global _epoch_used
+    if _epoch_used:
+        raise RuntimeError('set_epoch must be called before dates plotted.')
+    _epoch = epoch
+
+
+def get_epoch():
+    """
+    Get the epoch currently used by `.dates`.
```

Comment:
```suggestion
    Get the epoch used by `.dates`.
```
I don't think "currently" adds any information.